### PR TITLE
Add Redux support

### PR DESCRIPTION
**Pull request by rebel216**
init Redux and duck patterns.

### CODE REVIEW

**Review by Mihndim2020 — APPROVED** [review]
### Hi @rebel216, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._

### I will leave this comments here  because the files are not present for this pull request since you did not make any change on them.

### Your project is still missing the following requirements, please kindly add them 

Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

 - Define action types for adding and removing a book.
 - Set the initial state to be an empty array of books.
 - Export Action Creators for your actions.
 - Write your reducer and export it as default.
 - Define state changes for the actions that you created.
 - In case of unknown action - return the current state.
 - Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

 - Define an action type for checking the status.
 - Set the initial state to be an empty array of categories.
 - Export Action Creators for your actions.
 - Write your reducer and export it as default.
 - For the check status action return a string "Under construction".
 - In case of unknown action - return the current state.
 - Configure the Redux Store (/src/redux/configureStore.js):

 - Importing the necessary methods from Redux Toolkit.
 - Combine both reducers into a root reducer by using configureStore function.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Mihndim2020** [comment]
Hello @rebel216, please ***do not merge this pull request***, I  mistakenly approved it. Please do implement the requested changes and ask for another code review. Thank you.

File: src/redux/categories/categories.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+
+function Categories() {
+  return <div>Under construction</div>;
+}
+
+export default Categories;
```
Comment (phelian23) [inline src/redux/categories/categories.js]: - [ ] This file should contain your categories actions and reducer and not the Categories component, you already have that in the /src/components/Categories.js. and as requested by the previous reviewer, you should define an action type for checking the status, set the initial state to be an empty array of categories, export Action Creators for your actions, write your reducer and export it as default, and the check status action should return a string "Under construction".

**Review by phelian23 — CHANGES_REQUESTED** [review]
Hi @rebel216,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @rebel216 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR description 👍 
* Good looking readme file 👍 
* Proper use of Gitflow 👍 

 ## Required Changes ♻️
* As mentioned by the previous reviewer your categories lacks the necessary actions and reducers, just as you have done with the books, kindly do same for the categories 🙏 

> Define an action type for checking the status.
> Set the initial state to be an empty array of categories.
> Export Action Creators for your actions.
> Write your reducer and export it as default.
> For the check status action return a string "Under construction".
> In case of unknown action - return the current state.

_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Davidosky007 — COMMENTED** [review]
Hi @rebel216 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has failing linter checks kindly fix the linter error and resubmit for review 🙏 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @rebel216  😊 😃 
### [OPTIONAL]

- [ ] Please edit your README file according to your current project title and description with the tech stack used. 👍🏻 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**